src/bit_vec/tests.rs

```diff
@@ -213,14 +213,14 @@ fn test_custom_iter_behavior() {
     assert!(iter.advance_by(6).is_err());
     assert!(iter.advance_back_by(5).is_ok());
 
-    assert_eq!(bv.iter().skip(2).next(), Some(0));
```

Comment:
Please revert this and suppress the clippy correction. This intentionally calls skip() and next() to verify skip behaves correctly. This test should exist regardless of whether nth() internally uses skip (which I think it doesn't). Same for the change a few lines down.